translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+                <body>
+                    <trans-unit id="invitation_title">
+                        <source>invitation_title</source>
+                        <target>{organizer_gender, select, female {{organizer_name} has invited you for her party!} male   {{organizer_name} has invited you for his party!} other {{organizer_name} have invited you for their party!}}</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'invitation_title' => '{organizer_gender, select,
+                female {{organizer_name} has invited you for her party!}
+                male   {{organizer_name} has invited you for his party!}
+                other  {{organizer_name} have invited you for their party!}
+            }',
+        ];
+
+This might look very complex. The basic syntax for all functions is
+``{variable_name, function_name, function_statement}``. In this case, the
+function name is ``select`` and its statement contains the "cases" of this
+select. This function is applied over the ``organizer_gender`` variable::
+
+    // ...
+
+    // prints "Ryan has invited you for his party!"
+    echo $translator->trans('invition_title', [
+        'organizer_name' => 'Ryan',
+        'organizer_gender' => 'male',
+    ]);
+
+    // prints "John & Jane have invited you for their party!"
+    echo $translator->trans('invition_title', [
+        'organizer_name' => 'John & Jane',
+        'organizer_gender' => 'not_applicable',
+    ]);
+
+The ``{...}`` syntax alternates between "literal" and "code" mode. This allows
+you to use literal text in the select statements:
+
+#. The first ``{organizer_gender, select, ...}`` block starts the "code" mode,
+   which means ``organizer_gender`` is processed as a variable.
+#. The inner ``{... has invited you for her party!}`` block brings you back in
+   "literal" mode, meaning the text is not processed.
+#. Inside this block, ``{organizer_name}`` starts "code" mode again, allowing
+   ``organizer_name`` to be processed as variable.
+
+.. tip::
+
+    While it might seem more logical to only put ``her``, ``his`` or ``their``
+    in the switch statement, it is better to use "complex arguments" at the
+    outermost structure of the message. The strings are in this way better
+    readable for translators and, as you can see in the ``other`` case, other
+    parts of the sentence mgith be influenced by the variable.
+
+Pluralization
+-------------
+
+Another interesting function is ``plural``. It allows you to
+handle pluralization in your messages (e.g. ``There are 3 apples`` vs
+``There is one apple``). The function looks very similair to the ``select`` function:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        nr_of_apples: >
+            {apples, plural,
+                =0    {There are no apples :(}
```

Comment:
Let's remove the `:(` at the end of this phrase because it's confusing with the ICU syntax braces, etc.